### PR TITLE
Sleep duration in wait_until_true

### PR DESCRIPTION
**Pull request by teyamagu**
[proposal]

When I use `.wait_until_true` for long waits that don't need to be checked frequently, it overloads the execution environment with frequent yields and results in different behavior.

So I created a PR to solve the problem by using `sleep_duration` as an argument. What do you think of this idea?

There is an idea to make `sleep_duration` as follows for backward compatibility, but we did not adopt it because the way to define the argument is not simple. 

````
def self.wait_until_true(wait_time = Capybara.default_max_wait_time, sleep_duration: 0.05)
````

### CODE REVIEW

**Comment by luke-hill** [comment]
Not really sure I get the need for this, as it doesn't really affect your environment (the thread is super lightweight).

Can you add a unit test, then I can merge this. It doesn't really add anything bad so I figure it's ok to merge.

Typically if you're using a waiter and you want a lot more granularity, I would lean on the Selenium Wait class (As you can rescue errors and provide error messages).

**Comment by teyamagu** [comment]
Thank you for your comment.
I understand the general approach regarding `wait`.

I want to make this change, so I will add a unit test.

**Comment by teyamagu** [comment]
Hi @luke-hill ,

I added unit tests for sleep_duration. Could you check them?
If you have any questions, please feel free to let me know.

File: spec/site_prism/waiter_spec.rb [hunk 1]
```diff
@@ -23,5 +23,29 @@
...
+    context 'with a custom sleep_duration' do
+      let(:timeout) { 0.1 }
+      let(:sleep_duration_long) { 0.5 }
+      let(:sleep_duration_short) { 0.01 }
+
+      # rubocop:disable RSpec/MultipleExpectations
+      # rubocop:disable Style/Semicolon
+      it 'when setting sleep_duration > timeout, error raise and yield execute 2 times' do
+        count = 0
+        expect { described_class.wait_until_true(timeout, sleep_duration_long) { count += 1; false } }
```
Comment (luke-hill) [inline spec/site_prism/waiter_spec.rb:36]:
```
swallow_timeout do
  described_class.wait_until_true(timeout, sleep_duration_long) do
    count += 1
    false
  end
end
```

Comment (luke-hill) [reply spec/site_prism/waiter_spec.rb:36]: cf: https://github.com/site-prism/site_prism/blob/main/spec/spec_helper.rb#L59-L69

Comment (teyamagu) [reply spec/site_prism/waiter_spec.rb:36]: Thanks! I'll fix this.

Comment (teyamagu) [reply spec/site_prism/waiter_spec.rb]: I changed code.  https://github.com/site-prism/site_prism/pull/105/commits/e2c4728f4feb468edec730e0acb8835139be9b86

**Review by luke-hill — APPROVED** [review]

**Comment by luke-hill** [comment]
Cheers, Please add a changelog in the same format as done for other items in the top unreleased area (Don't amend version e.t.c.)

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 ### Added
 
 ### Changed
+- `Waiter.wait_until_true` can be set `sleep_duration`
```
Comment (luke-hill) [inline CHANGELOG.md:7]: `Waiter.wait_until_true` can now set an optional `sleep_duration`

Comment (teyamagu) [reply CHANGELOG.md]: Sorry to incorrect sentence. I fixed that https://github.com/site-prism/site_prism/pull/105/commits/e954e399351249e73f3e61ad94254d8fb2dd83c1

**Pull request merged**